diff -urN mysql-4.0.10-gamma.org/sql/mysqld.cc mysql-4.0.10-gamma/sql/mysqld.cc
--- mysql-4.0.10-gamma.org/sql/mysqld.cc	Wed Jan 29 12:29:20 2003
+++ mysql-4.0.10-gamma/sql/mysqld.cc	Wed Feb 12 02:04:17 2003
@@ -96,7 +96,13 @@
 #endif /* __WIN__ */
 
 #ifdef HAVE_LIBWRAP
+#define hosts_access hosts_access_old
+#define sock_host sock_host_old
+#define eval_client eval_client_old
 #include <tcpd.h>
+#undef hosts_access
+#undef sock_host
+#undef eval_client
 #include <syslog.h>
 #ifdef NEED_SYS_SYSLOG_H
 #include <sys/syslog.h>
@@ -105,6 +111,9 @@
 int deny_severity = LOG_WARNING;
 
 #ifdef __STDC__
+extern int hosts_access(struct request_info *);
+extern int sock_host(struct request_info *);
+extern int eval_client(struct request_info *);
 #define my_fromhost(A)	   fromhost(A)
 #define my_hosts_access(A) hosts_access(A)
 #define my_eval_client(A)  eval_client(A)
