--- mysql-5.1.21-beta/client/mysql_upgrade.c~	2007-08-29 19:02:11.000000000 +0300
+++ mysql-5.1.21-beta/client/mysql_upgrade.c	2007-08-29 22:38:46.000000000 +0300
@@ -36,7 +36,7 @@
 
 static my_bool opt_force, opt_verbose, debug_info_flag, debug_check_flag;
 static uint my_end_arg= 0;
-static char *opt_user= (char*)"root";
+static char *opt_user= (char*)"mysql";
 
 static DYNAMIC_STRING ds_args;
 
