diff -urN mysql-4.0.10-gamma.org/isam/Makefile.am mysql-4.0.10-gamma/isam/Makefile.am
--- mysql-4.0.10-gamma.org/isam/Makefile.am	Thu Feb 13 19:29:15 2003
+++ mysql-4.0.10-gamma/isam/Makefile.am	Thu Feb 13 19:29:26 2003
@@ -16,7 +16,7 @@
 
 INCLUDES =		@MT_INCLUDES@ -I$(srcdir)/../include -I../include 
 LDADD =			@CLIENT_EXTRA_LDFLAGS@ libnisam.a ../mysys/libmysys.a \
-			../dbug/libdbug.a ../strings/libmystrings.a
+			../dbug/libdbug.a ../strings/libmystrings.a $(libdir)/libsupc++.la
 pkglib_LIBRARIES =	libnisam.a
 bin_PROGRAMS =		isamchk isamlog pack_isam
 isamchk_DEPENDENCIES=	$(LIBRARIES)
diff -urN mysql-4.0.10-gamma.org/myisam/Makefile.am mysql-4.0.10-gamma/myisam/Makefile.am
--- mysql-4.0.10-gamma.org/myisam/Makefile.am	Wed Jan 29 12:29:22 2003
+++ mysql-4.0.10-gamma/myisam/Makefile.am	Thu Feb 13 19:32:55 2003
@@ -19,7 +19,7 @@
 
 INCLUDES =		@MT_INCLUDES@ -I$(srcdir)/../include -I../include
 LDADD =			@CLIENT_EXTRA_LDFLAGS@ libmyisam.a ../mysys/libmysys.a \
-			../dbug/libdbug.a ../strings/libmystrings.a
+			../dbug/libdbug.a ../strings/libmystrings.a $(libdir)/libsupc++.la
 pkglib_LIBRARIES =	libmyisam.a
 bin_PROGRAMS =		myisamchk myisamlog myisampack
 myisamchk_DEPENDENCIES=	$(LIBRARIES)
