--- mysql-5.1.21-beta/configure.in~	2007-08-29 19:00:29.000000000 +0300
+++ mysql-5.1.21-beta/configure.in	2007-08-29 19:00:35.000000000 +0300
@@ -836,8 +836,9 @@
 
 AC_CHECK_LIB(m, floor, [], AC_CHECK_LIB(m, __infinity))
 
-AC_CHECK_LIB(nsl_r, gethostbyname_r, [],
-  AC_CHECK_LIB(nsl, gethostbyname_r))
+AC_CHECK_FUNC(gethostbyname_r, [], [
+ AC_CHECK_LIB(nsl_r, gethostbyname_r, [],
+  AC_CHECK_LIB(nsl, gethostbyname_r)) ])
 AC_CHECK_FUNC(gethostbyname_r)
 
 AC_SEARCH_LIBS(setsockopt, socket)
@@ -845,6 +846,11 @@
 AC_SEARCH_LIBS(bind, bind)
 # Check if crypt() exists in libc or libcrypt, sets LIBS if needed
 AC_SEARCH_LIBS(crypt, crypt, AC_DEFINE(HAVE_CRYPT, 1, [crypt]))
+save_LIBS="$LIBS"
+AC_CHECK_LIB(crypt, crypt, [LIBS="-lcrypt $LIBS"; LIBCRYPT="-lcrypt"])
+AC_SUBST(LIBCRYPT)
+AC_CHECK_FUNC(crypt, AC_DEFINE([HAVE_CRYPT], [1], [crypt]))
+LIBS="$save_LIBS"
 # See if we need a library for address lookup.
 AC_SEARCH_LIBS(inet_aton, [socket nsl resolv])
 
--- mysql-5.1.21-beta/sql/Makefile.am~	2007-08-29 18:58:33.000000000 +0300
+++ mysql-5.1.21-beta/sql/Makefile.am	2007-08-29 19:01:33.000000000 +0300
@@ -43,7 +43,7 @@
 			@pstack_libs@ \
 			@mysql_plugin_libs@ \
 			$(LDADD)  $(CXXLDFLAGS) $(WRAPLIBS) @LIBDL@ \
-			$(yassl_libs) $(openssl_libs) @MYSQLD_EXTRA_LIBS@
+			$(yassl_libs) $(openssl_libs) @LIBCRYPT@ @MYSQLD_EXTRA_LIBS@
 
 noinst_HEADERS =	item.h item_func.h item_sum.h item_cmpfunc.h \
 			item_strfunc.h item_timefunc.h \
--- mysql-5.1.26-rc/config/ac-macros/ssl.m4.orig	2008-07-01 00:35:01.000000000 +0200
+++ mysql-5.1.26-rc/config/ac-macros/ssl.m4	2008-08-25 19:53:00.258254465 +0200
@@ -102,7 +102,12 @@
   #
   # Try to link with openSSL libs in <location>
   #
-  openssl_libs="-L$location/lib/ -lssl -lcrypto"
+  if test "$location" != "/usr"
+  then
+  openssl_libs="-L$location/lib -lssl -lcrypto"
+  else
+  openssl_libs="-lssl -lcrypto"
+  fi
   MYSQL_CHECK_SSL_DIR([$openssl_includes], [$openssl_libs])
 
   if test "$mysql_ssl_found" == "no"
