--- a/sql/mysqld.cc	2014-04-07 08:41:22.178686821 +0200
+++ b/sql/mysqld.cc	2014-04-07 08:46:10.282352356 +0200
@@ -6200,7 +6200,7 @@
             The connection was refused by TCP wrappers.
             There are no details (by client IP) available to update the host_cache.
           */
-          statistic_increment(connection_tcpwrap_errors, &LOCK_status);
+          statistic_increment(connection_errors_tcpwrap, &LOCK_status);
 	  continue;
 	}
       }
