--- mysql-5.1.14-beta/client/mysql_upgrade.c~	2006-12-13 12:09:28.860238973 +0200
+++ mysql-5.1.14-beta/client/mysql_upgrade.c	2006-12-13 12:10:17.331321518 +0200
@@ -40,7 +40,7 @@
 extern TYPELIB sql_protocol_typelib;
 
 static my_bool opt_force= 0, opt_verbose= 0, opt_compress= 0;
-static char *user= (char*) "root", *basedir= 0, *datadir= 0, *opt_password= 0;
+static char *user= (char*) "mysql", *basedir= 0, *datadir= 0, *opt_password= 0;
 static char *current_host= 0;
 static char *opt_default_charset= 0, *opt_charsets_dir= 0;
 #ifdef HAVE_SMEM
