--- mysql-5.1.6-alpha/sql-common/client.c~	2006-02-21 16:09:34.000000000 +0200
+++ mysql-5.1.6-alpha/sql-common/client.c	2006-02-21 16:14:10.000000000 +0200
@@ -1456,6 +1456,7 @@
 
   mysql->options.methods_to_use= MYSQL_OPT_GUESS_CONNECTION;
   mysql->options.report_data_truncation= TRUE;  /* default */
+  mysql_read_default_options(&mysql->options, "/etc/mysql/mysql-client.conf", NULL);
 
   /*
     By default we don't reconnect because it could silently corrupt data (after
