diff -burN mysql-4.0.14.orig/acinclude.m4 mysql-4.0.14/acinclude.m4
--- mysql-4.0.14.orig/acinclude.m4	Fri Jul 18 16:57:40 2003
+++ mysql-4.0.14/acinclude.m4	Thu Jul 24 23:22:42 2003
@@ -722,6 +722,9 @@
   if test -f $d/libssl.a ; then
     OPENSSL_LIB=$d
   fi
+  if test -f $d/libssl.so ; then
+    OPENSSL_LIB=$d
+  fi
  done
 
   # On RedHat 9 we need kerberos to compile openssl
diff -burN mysql-4.0.14.orig/configure.in mysql-4.0.14/configure.in
--- mysql-4.0.14.orig/configure.in	Fri Jul 18 16:57:46 2003
+++ mysql-4.0.14/configure.in	Thu Jul 24 23:29:00 2003
@@ -742,19 +742,21 @@
 
 AC_CHECK_LIB(m, floor, [], AC_CHECK_LIB(m, __infinity))
 
+AC_CHECK_FUNC(gethostbyname_r, , [
 AC_CHECK_LIB(nsl_r, gethostbyname_r, [],
-  AC_CHECK_LIB(nsl, gethostbyname_r))
-AC_CHECK_FUNC(gethostbyname_r)
+  AC_CHECK_LIB(nsl, gethostbyname_r))])
 
 AC_CHECK_FUNC(setsockopt, , AC_CHECK_LIB(socket, setsockopt))
-AC_CHECK_FUNC(yp_get_default_domain, ,
-  AC_CHECK_LIB(nsl, yp_get_default_domain))
 AC_CHECK_FUNC(p2open, , AC_CHECK_LIB(gen, p2open))
 # This may get things to compile even if bind-8 is installed
 AC_CHECK_FUNC(bind, , AC_CHECK_LIB(bind, bind))
 # For crypt() on Linux
-AC_CHECK_LIB(crypt, crypt)
+save_LIBS="$LIBS"
+LIBCRYPT=
+AC_CHECK_LIB(crypt, crypt, [LIBS="-lcrypt $LIBS"; LIBCRYPT="-lcrypt"])
+AC_SUBST(LIBCRYPT)
 AC_CHECK_FUNC(crypt, AC_DEFINE(HAVE_CRYPT)) 
+LIBS="$save_LIBS"
 
 # For sem_xxx functions on Solaris 2.6
 AC_CHECK_FUNC(sem_init, , AC_CHECK_LIB(posix4, sem_init))
diff -burN mysql-4.0.14.orig/libmysql_r/Makefile.am mysql-4.0.14/libmysql_r/Makefile.am
--- mysql-4.0.14.orig/libmysql_r/Makefile.am	Fri Jul 18 16:57:47 2003
+++ mysql-4.0.14/libmysql_r/Makefile.am	Thu Jul 24 23:30:25 2003
@@ -30,7 +30,7 @@
 libmysql_dir = $(top_srcdir)/libmysql
 
 libmysqlclient_r_la_SOURCES = $(target_sources)
-libmysqlclient_r_la_LIBADD = $(target_libadd)
+libmysqlclient_r_la_LIBADD = $(target_libadd) -lpthread
 libmysqlclient_r_la_LDFLAGS = $(target_ldflags)
 
 # This is called from the toplevel makefile
diff -burN mysql-4.0.14.orig/sql/Makefile.am mysql-4.0.14/sql/Makefile.am
--- mysql-4.0.14.orig/sql/Makefile.am	Fri Jul 18 16:57:41 2003
+++ mysql-4.0.14/sql/Makefile.am	Thu Jul 24 23:29:27 2003
@@ -43,7 +43,7 @@
 mysqld_LDADD =		@MYSQLD_EXTRA_LDFLAGS@ \
 			@bdb_libs@ @innodb_libs@ @pstack_libs@ \
                         @innodb_system_libs@ \
-			$(LDADD)  $(CXXLDFLAGS) $(WRAPLIBS) @LIBDL@ @openssl_libs@
+			$(LDADD)  $(CXXLDFLAGS) $(WRAPLIBS) @LIBDL@ @openssl_libs@ @LIBCRYPT@
 noinst_HEADERS =	item.h item_func.h item_sum.h item_cmpfunc.h \
 			item_strfunc.h item_timefunc.h item_uniq.h \
 			item_create.h mysql_priv.h \
diff -burN mysql-4.0.14.orig/vio/Makefile.am mysql-4.0.14/vio/Makefile.am
--- mysql-4.0.14.orig/vio/Makefile.am	Fri Jul 18 16:57:50 2003
+++ mysql-4.0.14/vio/Makefile.am	Thu Jul 24 23:24:27 2003
@@ -22,15 +22,15 @@
 test_ssl_SOURCES=	test-ssl.c
 test_ssl_LDADD=   	@CLIENT_EXTRA_LDFLAGS@ ../dbug/libdbug.a libvio.a \
 			../mysys/libmysys.a ../strings/libmystrings.a \
-			$(openssl_libs)
+			$(openssl_libs) ../libmysql_r/libmysqlclient_r.la
 test_sslserver_SOURCES=	test-sslserver.c
 test_sslserver_LDADD=   @CLIENT_EXTRA_LDFLAGS@	../dbug/libdbug.a libvio.a \
 			../mysys/libmysys.a ../strings/libmystrings.a \
-			$(openssl_libs)
+			$(openssl_libs) ../libmysql_r/libmysqlclient_r.la
 test_sslclient_SOURCES=   test-sslclient.c
 test_sslclient_LDADD=   @CLIENT_EXTRA_LDFLAGS@	../dbug/libdbug.a libvio.a \
 			../mysys/libmysys.a ../strings/libmystrings.a \
-			$(openssl_libs)
+			$(openssl_libs) ../libmysql_r/libmysqlclient_r.la
 libvio_a_SOURCES=	vio.c viosocket.c viossl.c viosslfactories.c
 
 # Don't update the files from bitkeeper
