--- mysql-5.1.22-rc/client/mysql_upgrade.c	2007-09-24 13:30:24.000000000 +0300
+++ mysql-5.1.22-rc-upgrade/client/mysql_upgrade.c	2008-01-12 17:28:56.945615880 +0200
@@ -31,12 +31,12 @@
 # endif
 #endif
 
-static char mysql_path[FN_REFLEN];
-static char mysqlcheck_path[FN_REFLEN];
+static char *mysql_path = "/usr/bin/mysql";
+static char *mysqlcheck_path = "/usr/sbin/mysqlcheck";
 
 static my_bool opt_force, opt_verbose, debug_info_flag, debug_check_flag;
 static uint my_end_arg= 0;
-static char *opt_user= (char*)"root";
+static char *opt_user= (char*)"mysql";
 
 static DYNAMIC_STRING ds_args;
 
@@ -758,12 +673,6 @@
   dynstr_append_os_quoted(&ds_args, "--user=", opt_user, NullS);
   dynstr_append(&ds_args, " ");
 
-  /* Find mysql */
-  find_tool(mysql_path, IF_WIN("mysql.exe", "mysql"));
-
-  /* Find mysqlcheck */
-  find_tool(mysqlcheck_path, IF_WIN("mysqlcheck.exe", "mysqlcheck"));
-
   /*
     Read the mysql_upgrade_info file to check if mysql_upgrade
     already has been run for this installation of MySQL
