--- mysql-4.1.12/Docs/mysql.info~	2005-05-16 08:05:46.000000000 +0300
+++ mysql-4.1.12/Docs/mysql.info	2005-05-16 08:06:36.000000000 +0300
@@ -1,8 +1,9 @@
 This is mysql.info, produced by makeinfo version 4.7 from manual.texi.
 
-START-INFO-DIR-ENTRY
-* mysql: (mysql).               MySQL documentation.
-END-INFO-DIR-ENTRY
+@dircategory Data bases:
+@direntry
+* mysql: (mysql).			MySQL documentation
+@end direntry
 
 
 File: mysql.info,  Node: Top,  Next: Introduction,  Prev: (dir),  Up: (dir)
