diff -urN mysql-4.0.9-gamma.org/acinclude.m4 mysql-4.0.9-gamma/acinclude.m4
--- mysql-4.0.9-gamma.org/acinclude.m4	Mon Feb  3 23:34:15 2003
+++ mysql-4.0.9-gamma/acinclude.m4	Tue Feb  4 00:11:55 2003
@@ -715,6 +715,9 @@
   if test -f $d/libssl.a ; then
     OPENSSL_LIB=$d
   fi
+  if test -f $d/libssl.so ; then
+    OPENSSL_LIB=$d
+  fi
  done
 
  if test -z "$OPENSSL_LIB" -o -z "$OPENSSL_INCLUDE" ; then

diff -urN mysql-4.0.10-gamma.org/vio/Makefile.am mysql-4.0.10-gamma/vio/Makefile.am
--- mysql-4.0.10-gamma.org/vio/Makefile.am	Wed Jan 29 12:29:24 2003
+++ mysql-4.0.10-gamma/vio/Makefile.am	Wed Feb 12 01:50:02 2003
@@ -21,13 +21,13 @@
 noinst_HEADERS =	
 test_ssl_SOURCES =   test-ssl.c
 test_ssl_LDADD =   	@CLIENT_EXTRA_LDFLAGS@ ../dbug/libdbug.a libvio.a ../mysys/libmysys.a \
-			../strings/libmystrings.a libvio.a $(openssl_libs)
+			../strings/libmystrings.a libvio.a $(openssl_libs) ../libmysql_r/libmysqlclient_r.la
 test_sslserver_SOURCES =   test-sslserver.c
 test_sslserver_LDADD =   @CLIENT_EXTRA_LDFLAGS@	../dbug/libdbug.a libvio.a ../mysys/libmysys.a \
-			../strings/libmystrings.a libvio.a $(openssl_libs)
+			../strings/libmystrings.a libvio.a $(openssl_libs) ../libmysql_r/libmysqlclient_r.la
 test_sslclient_SOURCES =   test-sslclient.c
 test_sslclient_LDADD =   @CLIENT_EXTRA_LDFLAGS@	../dbug/libdbug.a libvio.a ../mysys/libmysys.a \
-			../strings/libmystrings.a libvio.a $(openssl_libs)
+			../strings/libmystrings.a libvio.a $(openssl_libs) ../libmysql_r/libmysqlclient_r.la
 libvio_a_SOURCES =	vio.c viosocket.c viossl.c viosslfactories.c
 
 # Don't update the files from bitkeeper
