diff -urbB mysql-5.0.13-rc.org/libmysqld/client.c mysql-5.0.13-rc/libmysqld/client.c
--- mysql-5.0.13-rc.org/libmysqld/client.c	2005-09-22 21:40:24.000000000 +0200
+++ mysql-5.0.13-rc/libmysqld/client.c	2005-09-29 19:35:27.947950000 +0200
@@ -1455,6 +1455,7 @@
 
   mysql->options.methods_to_use= MYSQL_OPT_GUESS_CONNECTION;
   mysql->options.report_data_truncation= TRUE;  /* default */
+  mysql_read_default_options(&mysql->options, "/etc/mysql/mysql-client.conf", NULL);
   return mysql;
 }
 
diff -urbB mysql-5.0.13-rc.org/sql/client.c mysql-5.0.13-rc/sql/client.c
--- mysql-5.0.13-rc.org/sql/client.c	2005-09-22 21:40:24.000000000 +0200
+++ mysql-5.0.13-rc/sql/client.c	2005-09-29 19:35:38.164588500 +0200
@@ -1455,6 +1455,7 @@
 
   mysql->options.methods_to_use= MYSQL_OPT_GUESS_CONNECTION;
   mysql->options.report_data_truncation= TRUE;  /* default */
+  mysql_read_default_options(&mysql->options, "/etc/mysql/mysql-client.conf", NULL);
   return mysql;
 }
 
diff -urbB mysql-5.0.13-rc.org/sql-common/client.c mysql-5.0.13-rc/sql-common/client.c
--- mysql-5.0.13-rc.org/sql-common/client.c	2005-09-22 21:40:24.000000000 +0200
+++ mysql-5.0.13-rc/sql-common/client.c	2005-09-29 19:35:54.881633250 +0200
@@ -1455,6 +1455,7 @@
 
   mysql->options.methods_to_use= MYSQL_OPT_GUESS_CONNECTION;
   mysql->options.report_data_truncation= TRUE;  /* default */
+  mysql_read_default_options(&mysql->options, "/etc/mysql/mysql-client.conf", NULL);
   return mysql;
 }
 
